models/seir/model_mod.f90

```diff
@@ -0,0 +1,455 @@
+! time you want the model to advance between assimilations.
+! This interface is required for all applications.
+
+function shortest_time_between_assimilations()
+
+type(time_type) :: shortest_time_between_assimilations
+
+shortest_time_between_assimilations = time_step
+
+end function shortest_time_between_assimilations
+
+
+
+!------------------------------------------------------------------
+! Given a state handle, a location, and a model state variable quantity,
+! interpolates the state variable fields to that location and returns
+! the values in expected_obs. The istatus variables should be returned as
+! 0 unless there is some problem in computing the interpolation in
+! which case an alternate value should be returned. The itype variable
+! is an integer that specifies the quantity of field (for
+! instance temperature, zonal wind component, etc.). In low order
+! models that have no notion of types of variables this argument can
+! be ignored. For applications in which only perfect model experiments
+! with identity observations (i.e. only the value of a particular
+! state variable is observed), this can be a NULL INTERFACE.
+
+subroutine model_interpolate(state_handle, ens_size, location, iqty, expected_obs, istatus)
+
+type(ensemble_type),  intent(in) :: state_handle
+integer,              intent(in) :: ens_size
+type(location_type),  intent(in) :: location
+integer,              intent(in) :: iqty
+real(r8),            intent(out) :: expected_obs(ens_size) !< array of interpolated values
+integer,             intent(out) :: istatus(ens_size)
+
+! Given the nature of the SEIR model, no interpolation 
+! is needed. Only identity observations are utilized.
+
+! This should be the result of the interpolation of a
+! given quantity (iqty) of variable at the given location.
+expected_obs(:) = MISSING_R8
+
+! The return code for successful return should be 0. 
+! Any positive number is an error.
+! Negative values are reserved for use by the DART framework.
+! Using distinct positive values for different types of errors can be
+! useful in diagnosing problems.
+istatus(:) = 1
+
+end subroutine model_interpolate
+
+
+
+!------------------------------------------------------------------
+! Given an integer index into the state vector structure, returns the
+! associated location. A second intent(out) optional argument quantity
+! (qty) can be returned if the model has more than one type of field
+! (for instance temperature and zonal wind component). This interface is
+! required for all filter applications as it is required for computing
+! the distance between observations and state variables.
+
+subroutine get_state_meta_data(index_in, location, qty_type)
+
+integer(i8),         intent(in)  :: index_in
+type(location_type), intent(out) :: location
+integer,             intent(out), optional :: qty_type
+
+! these should be set to the actual location and state quantity
+location = state_loc(index_in)
```

Comment:
The location is fixed and 0.5 for all state variables, so you can set that here.


```suggestion
! The SEIR variables have no physical location, 
! and so I'm placing all 7 variables at the same 
! virtual point in space.
x_loc = 0.5_r8
```

Reply:
edit: sorry this suggestion should be

location = state_loc